### PR TITLE
EventGrid: Update README files to include the current new preview api…

### PR DESCRIPTION
**Pull request by ahamad-MS**
… version 2018-09-preview

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [X] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/eventgrid/resource-manager/readme.md [hunk 1]
```diff
@@ -159,12 +159,22 @@ go:
...
 ``` yaml $(go) && $(multiapi)
 batch:
+  - tag: package-2018-09-preview
   - tag: package-2018-05-preview
   - tag: package-2018-01
   - tag: package-2017-09-preview
   - tag: package-2017-06-preview
 ```
 
+### Tag: package-2018-09-preview and go
```
Comment (kalyanaj) [inline specification/eventgrid/resource-manager/readme.md]: Looking at the latest readme.me in the master, it looks like the Go configuration has been branched out to a separate file. It looks like you need to sync those changes in your branch and this change needs to happen in the GO specific readme file.

**Comment by kalyanaj** [comment]
The linter error is a known issue in the ARM validation: https://github.com/Azure/azure-openapi-arm-validator/issues/10.

**Comment by praries880** [comment]
@azuresdkci add to whitelist

**Review by kalyanaj — APPROVED** [review]
LGTM.

**Comment by kalyanaj** [comment]
@praries880, the "Automation for azure-sdk-for-go" CI step seems to be failing, however reviewing the readme.go.md file, the keys seem to be unique and we can't find what the issue is. Can you please help us with this issue?

**Comment by praries880** [comment]
@ahamad-MS you need to fix the ARM validation error : https://github.com/Azure/azure-rest-api-specs/pull/4235/checks?check_run_id=23938472 to get green on the CI

Heres the error 👍 

  | R3026 - TrackedResourcePatchOperation | 2018-09-15-preview/​EventGrid.json:2134 | Tracked resource 'Domain' must have patch operation that at least supports the update of tags. It's strongly recommended that the PATCH operation supports update of all mutable properties as well.
-- | -- | -- | --




@veronicagg Do you know why was the linter error in the json not caught earlier?

**Comment by kalyanaj** [comment]
@praries880, please see my earlier comment above: the linter error is a known issue in the ARM validation: Azure/azure-openapi-arm-validator#10 and is being tracked by the ARM team.

File: specification/eventgrid/resource-manager/readme.go.md [hunk 2]
```diff
@@ -13,11 +13,19 @@ go:
...
   - tag: package-2017-09-preview
   - tag: package-2017-06-preview
 ```
+### Tag: package-2018-09-preview and go
+
+These settings apply only when `--tag=package-2018-09-preview --go` is specified on the command line.
+Please also specify `--go-sdk-folder=<path to the root directory of your azure-sdk-for-go clone>`.
+
+``` yaml $(tag) == 'package-2018-09-preview' && $(go)
+output-folder: $(go-sdk-folder)/services/preview/$(namespace)/mgmt/2018-09-15-preview/$(namespace)
```
Comment (jhendrixMSFT) [inline specification/eventgrid/resource-manager/readme.go.md]: The closing ``` is missing causing a parsing failure.

Comment (ahamad-MS) [reply specification/eventgrid/resource-manager/readme.go.md]: @jhendrixMSFT  thanks for the help. this seems to fix the parsing issue.

Comment (ahamad-MS) [reply specification/eventgrid/resource-manager/readme.go.md]: @praries880 We are down to the known linter error/failure here only (https://github.com/Azure/azure-openapi-arm-validator/issues/10)

**Review by praries880 — APPROVED** [review]

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**